command/format.go

```diff
@@ -0,0 +1,160 @@
+package command
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/hcl/v2/hclparse"
+	"github.com/hashicorp/hcl/v2/hclwrite"
+	hclutils "github.com/hashicorp/packer/hcl2template"
+	"github.com/posener/complete"
+)
+
+const (
+	hcl2FileExt        = ".pkr.hcl"
+	hcl2JsonFileExt    = ".pkr.json"
+	hcl2VarFileExt     = ".auto.pkrvars.hcl"
+	hcl2VarJsonFileExt = ".auto.pkrvars.json"
+)
+
+type FormatCommand struct {
+	Meta
+	parser *hclparse.Parser
+}
+
+func (c *FormatCommand) Run(args []string) int {
+	ctx := context.Background()
+	cfg, ret := c.ParseArgs(args)
+	if ret != 0 {
+		return ret
+	}
+
+	// Might need to check for cfg type.
+	return c.RunContext(ctx, cfg)
+}
+
+func (c *FormatCommand) ParseArgs(args []string) (*FormatArgs, int) {
+	var cfg FormatArgs
+	flags := c.Meta.FlagSet("format", FlagSetNone)
+	flags.Usage = func() { c.Ui.Say(c.Help()) }
+	cfg.AddFlagSets(flags)
+	if err := flags.Parse(args); err != nil {
+		return &cfg, 1
+	}
+
+	// What is Path here?
```

Comment:
did you ever find the answer?

Reply:
It's the path to what we want Packer to see, so for example: `.` in `packer build .`

Reply:
I've hesitated allowing to pass multiple files here and still think that only allowing folder or single file is the best way to go for now.

Reply:
@SwampDragons yes, I'll drop the comment. 

> I've though about allowing to pass multiple files here and still think that only allowing folder or single file is the best way to go.

@azr totally agree. I think matching the behavior of `packer build` is a good way forward as it keeps file parsing consistent. With that said, I need to change the condition a bit to ensure only the first non-flag argument gets selected.